### DISCUSSION TITLE
rapid drop/reseed

### DISCUSSION THREAD

**Discussion by reharik** [prisma/prisma#9665] [Q&A]
Hi, I would like to be able to drop the db, recreate it and run the seed as rapidly as possible.  Right now I'm not even sure if I'm getting it to drop consistently.  The need is to clean the db between cypress ( e2e ) tests.  So speed of the essence.  
The command I'm running is `prisma migrate reset --force`.  Is there any faster or better way to do this?  Does anyone know enough about these commands to tell me what they are doing and if I might be able to build a script that did less?
Thanks,
Raif

**Comment by ryands17** [comment]
@reharik 👋 

Yes `prisma migrate reset` is the only way to perform this in Prisma as of now. What it does would be dropping the database, applying the migrations and running the seed file.

**Reply by reharik** [reply]
So is there anything else it does? I think it regenerates the client code?  I mean if there are things in there I don't need, then I could perhaps look at the code and see and maybe submit a pr etc.  Perhaps if the code is somewhat centrally located then you could point me to the correct file? 
Thanks a lot!
Raif

**Reply by janpio** [reply]
From `npx prisma migrate reset --help`:

```
  --skip-generate   Skip triggering generators (e.g. Prisma Client)
      --skip-seed   Skip triggering seed
```

**Reply by reharik** [reply]
So I don't really know the internals of prisma, so if you recreate the db like that and you skip generating the client, but you have not made any schema changes, should the api be ok?

**Reply by janpio** [reply]
That depends on your application and what you _need_. 
It might be faster to drop and recreate the database via raw queries (possibly outside of Prisma Client).

**Reply by reharik** [reply]
yes. that makes sense.  Thanks for the help folks!
Raif

**Reply by amcjen** [reply]
For what it's worth, we just run `npx prisma db seed` in our `before()` Cypress test specs.  We don't need the entire database rebuilt every test, we just need clean, standardized data within the database. 

To do that, our seed file is actually just SQL that we run, so it's easy to populate with whatever you need. Our `seed.ts` file is just a bash file that calls the `mysql <dbname> < seed.sql` command. 

Works great, and each reset only takes maybe a second or so. (Only gotcha is to make sure to add `TRUNCATE tablename` before each insert so you have a clean table to work from.)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
